satellite/console/service.go

```diff
@@ -119,22 +119,41 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 		return nil, errs.New(internalErrMsg)
 	}
 
-	u, err = s.store.Users().Insert(ctx, &User{
-		Email:        user.Email,
-		FullName:     user.FullName,
-		ShortName:    user.ShortName,
-		PasswordHash: hash,
-	})
+	// store data
+	tx, err := s.store.BeginTx(ctx)
 	if err != nil {
-		return nil, errs.New(internalErrMsg)
+		return nil, err
 	}
 
-	err = s.store.RegistrationTokens().UpdateOwner(ctx, registrationToken.Secret, u.ID)
-	if err != nil {
-		return nil, errs.New(internalErrMsg)
-	}
+	// clean up user on error
+	defer func() {
+		if err != nil {
+			u = nil
+		}
+	}()
+
+	err = withTx(tx, func(tx DBTx) error {
+		u, err = tx.Users().Insert(ctx,
+			&User{
+				Email:        user.Email,
+				FullName:     user.FullName,
+				ShortName:    user.ShortName,
+				PasswordHash: hash,
+			},
+		)
+		if err != nil {
+			return errs.New(internalErrMsg)
+		}
+
+		err = tx.RegistrationTokens().UpdateOwner(ctx, registrationToken.Secret, u.ID)
+		if err != nil {
+			return errs.New(internalErrMsg)
+		}
+
+		return nil
+	})
```

Comment:
why is this create user stuff changing in this PR at all? isn't this PR about stripe?

one thing you can do to break up commits is use `git add -p` which is very helpful. it allows you to add only some changes in a file. is this change stripe related?

Reply:
as this is only part of original PR, those changes were added later with more stripe related stuff. And when picking changes to commit I accidently picked them up to this PR